### PR TITLE
Adds `/` api to check if the server has started successfully

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
Endpoint: /
  
 - This api will return status as "ok" when server has started successfully

### CODE REVIEW

File: api/design/ok.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package design
+
+import (
+	. "goa.design/goa/v3/dsl"
+)
+
+var _ = Service("ok", func() {
```
Comment (SM43) [inline api/design/ok.go:7]: Should we rename it to `status` service?

File: api/design/ok.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	Description("Returns status ok")
```
Comment (SM43) [inline api/design/ok.go:8]: Describes the status of the server?

File: api/design/ok.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+	Error("internal-error", ErrorResult, "Internal Server Error")
+	Error("not-found", ErrorResult, "Resource Not Found Error")
```
Comment (SM43) [inline api/design/ok.go:11]: `not-found` ? we are returning any resources. so there won't be any error

File: api/design/ok.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+				})
+
+				Required("status")
+			})
+
+		HTTP(func() {
+			GET("/")
+
+			Response(StatusOK)
+			Response("internal-error", StatusInternalServerError)
```
Comment (SM43) [inline api/design/ok.go:29]: Only `StatusOK` willl be needed

File: api/go.mod [hunk 5]
```diff
@@ -8,11 +8,13 @@ require (
 	github.com/jinzhu/gorm v1.9.12
 	github.com/joho/godotenv v1.3.0
 	github.com/lib/pq v1.3.0
+	github.com/prometheus/common v0.10.0
```
Comment (SM43) [inline api/go.mod:11]: are we using `prometheus`? you can run `go mod tidy`. It will remove unused dependencies

File: api/pkg/service/ok/ok.go [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+	ok "github.com/tektoncd/hub/api/gen/ok"
+	"github.com/tektoncd/hub/api/pkg/app"
+	"go.uber.org/zap"
+)
+
+// ok service example implementation.
+// The example methods log the requests and return zero values.
+type service struct {
+	logger *zap.SugaredLogger
```
Comment (SM43) [inline api/pkg/service/ok/ok.go:14]: No need of `logger` as it won't be used here

File: api/pkg/service/ok/ok_test.go [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/tektoncd/hub/api/pkg/testutils"
+)
+
+func TestOk(t *testing.T) {
```
Comment (SM43) [inline api/pkg/service/ok/ok_test.go:11]: I think we can skip this test, Only HTTP test would be enough to test status code and body..

File: api/pkg/service/ok/testdata/TestOk_http.golden [hunk 8]
```diff
@@ -0,0 +1,3 @@
+{
+	"status": "ok"
```
Comment (SM43) [inline api/pkg/service/ok/testdata/TestOk_http.golden:2]: no need of golden file u can convert to map and check the value of the field

File: api/cmd/api/main.go [hunk 9]
```diff
@@ -51,21 +53,25 @@ func main() {
...
+		statusSvc   status.Service
 	)
 	{
 		categorySvc = categorysvc.New(api)
 		resourceSvc = resourcesvc.New(api)
+		statusSvc = status_svc.New()
 	}
 
-	// Wrap the services in endpoints that can be invoked from other services
+	// Wrap the services in endpoints that can be invstatused from other services
```
Comment (sthaha) [inline api/cmd/api/main.go:64]: nit: typo

File: api/pkg/service/status/status_http_test.go [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+		defer r.Body.Close()
+
+		var jsonMap map[string]interface{}
+		marshallErr := json.Unmarshal([]byte(b), &jsonMap)
+		assert.NoError(t, marshallErr)
+
+		assert.Equal(t, "ok", jsonMap["status"])
+
+	})
+}
```
Comment (sthaha) [inline api/pkg/service/status/status_http_test.go:37]: could you please add readiness and livenessProbe as well 🙏

File: api/pkg/service/status/status.go [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+import (
+	"context"
+
+	"github.com/tektoncd/hub/api/gen/status"
+	"go.uber.org/zap"
+)
+
+// status service example implementation.
+// The example methods log the requests and return zero values.
```
Comment (sthaha) [inline api/pkg/service/status/status.go:11]: this isn't true :)

File: api/cmd/api/http.go [hunk 12]
```diff
@@ -17,15 +17,18 @@ import (
 	category "github.com/tektoncd/hub/api/gen/category"
 	categorysvr "github.com/tektoncd/hub/api/gen/http/category/server"
 	resourcesvr "github.com/tektoncd/hub/api/gen/http/resource/server"
+	status_svr "github.com/tektoncd/hub/api/gen/http/status/server"
```
Comment (sthaha) [inline api/cmd/api/http.go:20]: statussvr to follow the same naming convention 🙏

File: api/cmd/api/main.go [hunk 13]
```diff
@@ -105,7 +111,7 @@ func main() {
 			} else if u.Port() == "" {
 				u.Host += ":80"
 			}
-			handleHTTPServer(ctx, u, &wg, errc, *dbgF, categoryEndpoints, resourceEndpoints, logger)
+			handleHTTPServer(ctx, u, &wg, errc, *dbgF, categoryEndpoints, resourceEndpoints, statusEndpoints, logger)
```
Comment (sthaha) [inline api/cmd/api/main.go:114]: NIT: could you please break the long time 
```go 
handleHTTPServer(
  ctx, u, &wg, err, 
  endpoints ... 
)
```
👼

File: api/design/status.go [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+
+var _ = Service("status", func() {
+	Description("Describes the status of the server")
+
+	Method("Status", func() {
+		Description("Return status 'ok' when the server is started successfully")
+
+		Result(
+			func() {
+				Attribute("status", String, "Default stautus of route", func() {
```
Comment (sthaha) [inline api/design/status.go:15]: typo: status

File: api/design/status.go [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+
+import (
+	. "goa.design/goa/v3/dsl"
+)
+
+var _ = Service("status", func() {
+	Description("Describes the status of the server")
+
+	Method("Status", func() {
+		Description("Return status 'ok' when the server is started successfully")
```
Comment (sthaha) [inline api/design/status.go:11]: is -> has: when the server has

File: api/pkg/service/status/status.go [hunk 16]
```diff
@@ -0,0 +1,24 @@
+package status
+
+import (
+	"context"
+
+	"github.com/tektoncd/hub/api/gen/status"
+)
+
+// status service example implementation.
```
Comment (sthaha) [inline api/pkg/service/status/status.go:9]: not an example implementation :D is it?

File: api/config/20-api-deployment.yaml [hunk 17]
```diff
@@ -20,6 +20,24 @@ spec:
           image: ko://github.com/tektoncd/hub/api/cmd/api
           ports:
             - containerPort: 8000
+          readinessProbe:
```
Comment (SM43) [inline api/config/20-api-deployment.yaml:23]: Could you please add `probes` in DB deployment also?

Comment (sthaha) [reply api/config/20-api-deployment.yaml:23]: @SM43 that is unrelated to this patch perhaps another issue/bug.

**Pull request merged**